### PR TITLE
samples: net: google_iot_mqtt: Change read publish to blocking 

### LINKED ISSUE #40965: Halt on receipt of Google Cloud IoT Core MQTT message sized 648+ bytes
**Describe the bug**

I adapted the [google_iot_mqtt sample](https://github.com/zephyrproject-rtos/zephyr/tree/main/samples/net/cloud/google_iot_mqtt) to the Nordic nRF9160-DK dev board. I found that when setting the Google Cloud IoT Core config payload to content of approximately 648 bytes or larger, the Zephyr application halts and becomes unresponsive. If shell is enabled, the shell halts and becomes unresponsive as well. It does not output any error on the console or reset. This behavior become more consistent with higher payload sizes, such as 768 bytes. Note that these payloads are smaller than the rx_buffer, which is left at the default 1024. See also [my issue on my repo](https://github.com/JeffreyUrban/nRF9160-google_iot_mqtt/issues/2).

In debugging, I found that this halt occurs before generating log output from entering the corresponding MQTT_EVT_PUBLISH event in protocol.c, so this appears to be happening in the Zephyr MQTT library. See `zephyr/subsys/net/lib/mqtt/mqtt_rx.c` `evt.type = MQTT_EVT_PUBLISH`.

Perhaps there is a difference (possible MQTT non-compliance) on the Google Cloud IoT Core side that is not accommodated by Zephyr's MQTT library.

I do not have a target that is compatible with the original example, so I cannot confirm if this issue affects the original sample. I could use some assistance or suggestions to troubleshoot further.

**To Reproduce**
1. `mkdir workspace`
2. `cd workspace`
3. `git clone git@github.com:JeffreyUrban/nRF9160-google_iot_mqtt.git` 
4. Create `nRF9160-google_iot_mqtt/west.yml`:
```
manifest:
  remotes:
    - name: ncs
      url-base: https://github.com/nrfconnect
  projects:
    - name: sdk-nrf
      remote: ncs
      path: nrf
      revision: v1.7.1
      import: true
      clone-depth: 1
  self:
    path: nRF9160-google_iot_mqtt
```
5. `west init -l nRF9160-google_iot_mqtt`
6. `west update`
7. (Set up GCP IoT Core, download certs and program to the nRF9160-DK, set GCP values in prj.conf... see [README.rst](https://github.com/JeffreyUrban/nRF9160-google_iot_mqtt) for full detail. Optionally, add `CONFIG_SHELL=y`)
8. `cd nRF9160-google_iot_mqtt`
9. `west build -b nrf9160dk_nrf9160_ns`
10. (plug in by USB and power on dev board)
11. `west flash --erase`
12. Boot and monitor the console via picocom while entering a config message in Google Cloud IoT Core console. Or, enter a config message in Google Cloud IoT Core console, and then boot and monitor the console via picocom.

**Expected behavior**

In experimenting with other similar samples using the Zephyr MQTT library, such as in the nRF Connect SDK mqtt_simple sample, and on different MQTT brokers (mqtt.eclipseprojects.io, test.mosquitto.org), larger packet sizes are received OK, or generate an error publish_get_payload failed: -122 when the payload exceeds the buffer size, and the application disconnects and reconnects to the MQTT broker. I expected similar behavior.

**Impact**
What impact does this issue have on your progress (e.g., annoyance, showstopper)

This prevents use of Google Cloud IoT Core as an IoT backend.

**Logs and console output**

None. The console becomes completely unresponsive before any output corresponding to the message.

**Environment (please complete the following information):**
 - OS: MacOS
 - Toolchain nRF v1.7.1 via west
 - Commit SHA or Version used: nRF v1.7.1

**Additional context**

Original and unchanged nRF9160-DK dev board.

### PR DESCRIPTION
**Pull request by JeffreyUrban**
Avoid endless loop in `case MQTT_EVT_PUBLISH`

Fixes #40965

Signed-off-by: Jeffrey Urban <JeffreyUrban@users.noreply.github.com>

### CODE REVIEW

File: samples/net/cloud/google_iot_mqtt/src/protocol.c [hunk 1]
```diff
@@ -166,7 +166,7 @@ void mqtt_evt_handler(struct mqtt_client *const client,
 
 		/* assuming the config message is textual */
 		while (len) {
-			bytes_read = mqtt_read_publish_payload(
+			bytes_read = mqtt_read_publish_payload_blocking(
 				client, d,
 				len >= 32 ? 32 : len);
 			if (bytes_read < 0 && bytes_read != -EAGAIN) {
```
Comment (rlubos) [inline samples/net/cloud/google_iot_mqtt/src/protocol.c:172]: I think we don't need to check for `bytes_read != -EAGAIN` here either, just break on any error.

**Review by rlubos — COMMENTED** [review]
Thanks, we don't put the PR number in the commit titles though, if you add the follwoing to the commit body it shall automatically close the issue on merge:
```
Fixes #40965
```
Also, please use the following prefix in the title: `samples: net: google_iot_mqtt: `.

**Comment by carlescufi** [comment]
> Signed-off-by: Jeffrey Urban [JeffreyUrban@users.noreply.github.com](mailto:JeffreyUrban@users.noreply.github.com)

You have to use a real email address.

**Comment by JeffreyUrban** [comment]
I made the suggested changes.

**Review by rlubos — COMMENTED** [review]
The second commit message exceeds the char/line limit (see the compliance result), this needs fixing (or just squash the commits, IMO there's no need for a separate ones).

I'm not sure about the other failures, @carlescufi any clue what's wrong in there (twister/clang)?

**Comment by carlescufi** [comment]
> I'm not sure about the other failures, @carlescufi any clue what's wrong in there (twister/clang)?

No, can you rebase your branch on top of the latest `main` @JeffreyUrban?

**Comment by carlescufi** [comment]
@JeffreyUrban 

```
Commit 5baa5f905d:
3: UC4 Line exceeds max length (80>75): "Fail on any error for mqtt_read_publish_payload_blocking, not excluding -EAGAIN."
```

**Comment by JeffreyUrban** [comment]
Thank you for being patient. I rebased and squashed and set a commit-msg gitlint hook. It appears to be still failing clang/twister, and I don't know why.

**Review by rlubos — APPROVED** [review]

**Comment by rlubos** [comment]
@JeffreyUrban The CI faliures observed in this PR are not related to your changes, but rather to faulty `sample.yaml` defined for the sample. I've submitted #41103 to address the problem and will ask you to rebase once more after it gets merged.

**Review by jukkar — APPROVED** [review]

**Pull request merged**